vault/identity_store_oidc.go

```diff
@@ -472,6 +472,25 @@ func (i *IdentityStore) pathOIDCCreateUpdateKey(ctx context.Context, req *logica
 		return logical.ErrorResponse("verification_ttl cannot be longer than 10x rotation_period"), nil
 	}
 
+	if req.Operation == logical.UpdateOperation {
+		// ensure any roles referencing this key do not already have a token_ttl
+		// greater than the key's verification_ttl
+		rolesReferencingTargetKeyName, err := i.getRolesReferencingTargetKeyName(ctx, req, name)
+		if err != nil {
+			return nil, err
+		}
+		for _, role := range rolesReferencingTargetKeyName {
+			if role.TokenTTL > key.VerificationTTL {
```

Comment:
I'm fine with this approach, though this doesn't guard against existing role and key values unless they are updated manually.